clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -319,4 +319,65 @@ public void createServerGroup_ArtifactsFARGATETgMappingsTest()
     assertEquals(80, serviceLB.getContainerPort().intValue());
     assertEquals("integArtifactsFargateTgMappings-cluster", seenCreateServRequest.getCluster());
   }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ task def artifacts, EC2 launch type, and new target group fields "
+          + "without container definition, gives an exception(Provided task definition does not contain any container definitions). "
+          + "\n===")
+  @Test
+  public void createServerGroup_ArtifactsEC2TgMappingsWithNoContainersTest()
+      throws IOException, InterruptedException {
+
+    // given
+    String url = getTestUrl(CREATE_SG_TEST_PATH);
+    String requestBody =
+        generateStringFromTestFile("/createServerGroup-artifact-EC2-targetGroupMappings.json");
+    String expectedServerGroupName = "ecs-integArtifactsEC2TgMappingsStack-detailTest";
+
+    setEcsAccountCreds();
+
+    ByteArrayInputStream byteArrayInputStreamOfArtifactsForEC2Type =
+        new ByteArrayInputStream(
+            generateStringFromTestArtifactFile(
+                    "createServerGroup-artifact-EC2-targetGroup-WithNoContainers-artifactFile.json")
+                .getBytes());
+
+    when(mockArtifactDownloader.download(any(Artifact.class)))
+        .thenReturn(byteArrayInputStreamOfArtifactsForEC2Type);
+
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    retryUntilTrue(
+        () -> {
+          List<Object> taskHistory =
+              get(getTestUrl("/task/" + taskId))
+                  .then()
+                  .contentType(ContentType.JSON)
+                  .extract()
+                  .path("history");
+          if (taskHistory
+              .toString()
+              .contains(
+                  String.format(
+                      "Provided task definition does not contain any container definitions"))) {
```

Comment:
I kind of like asserting on this exception text b/c it indicates the expected condition (no containers in artifact) was correctly detected and messaged to the end user. However, it's not a hard guarantee that the task _itself_ is [in a failed state](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java#L363), which is the expected deployment behavior. 

Is it possible to assert on the presence of the exception text in the response "status" obj, _and_ return the value of `status.failed` like we do for the other error case?

Reply:
Added assertion on status value that shows the presence of the exception text in the response.